pyhealth/datasets/base_dataset.py

```diff
@@ -309,6 +308,42 @@ def set_task(
                 self.patients.items(), desc=f"Generating samples for {self.task}"
         ):
             samples.extend(self.task_fn(patient))
+
+        # validate the samples
+        assert all(isinstance(s, dict) for s in samples), "Each sample should be a dict"
+        keys = samples[0].keys()
+        assert all(set(s.keys()) == set(keys) for s in samples), \
+            "All samples should have the same keys"
+        assert "patient_id" in keys, "patient_id should be in the keys"
+        assert "visit_id" in keys, "visit_id should be in the keys"
+        for key in keys:
+            # either a single value, a list of values, or a list of list of values
+            check = is_homo_list([s[key] for s in samples])
```

Comment:
I just jumped to `is_homo_list`. It would be good to have "Example" for this function.

Reply:
resolved in the latest commit